src/batch/src/rpc/service/task_service.rs

```diff
@@ -101,7 +109,60 @@ impl TaskService for BatchServiceImpl {
             Ok(_) => Ok(Response::new(RemoveTaskResponse { status: None })),
             Err(e) => {
                 error!("failed to remove task {}", e);
-                Err(e.to_grpc_status())
+                Err(e.into())
+            }
+        }
+    }
+
+    #[cfg_attr(coverage, no_coverage)]
+    async fn execute(
+        &self,
+        req: Request<ExecuteRequest>,
+    ) -> Result<Response<Self::ExecuteStream>, Status> {
+        let peer_addr = req
+            .remote_addr()
+            .ok_or_else(|| Status::unavailable("connection unestablished"))?;
+        let req = req.into_inner();
+        let task_id = req.get_task_id().expect("no task id found");
+        let plan = req.get_plan().expect("no plan found").clone();
+        let epoch = req.epoch;
+        let context = ComputeNodeContext::new(self.env.clone());
+        let task = BatchTaskExecution::new(task_id, plan, context, epoch)?;
+        let task = Arc::new(task);
+        if let Err(e) = task.clone().async_execute().await {
+            error!(
+                "failed to build executors and trigger execution of Task {:?}: {}",
+                task_id, e
+            );
+            return Err(e.into());
+        }
+
+        let pb_task_output_id = TaskOutputId {
+            task_id: Some(task_id.clone()),
+            // Since this is local execution path, the exchange would follow single distribution,
+            // therefore we would only have one data output.
+            output_id: 0,
+        };
+
+        match task.get_task_output(&pb_task_output_id) {
+            Ok(output) => {
+                let (tx, rx) = tokio::sync::mpsc::channel(LOCAL_EXECUTE_BUFFER_SIZE);
+                // if let Err(e) = self.mgr.get_data(tx, peer_addr, &pb_task_output_id) {
+                //     error!(
+                //         "Failed to serve local execution's exchange RPC from {}: {}",
+                //         peer_addr, e
+                //     );
+                //     return Err(e.to_grpc_status());
+                // }
+                //
+                Ok(Response::new(ReceiverStream::new(rx)))
```

Comment:
Remove unused comments?

Reply:
Sure!

Actually haven't finished yet, the `buf lint` forces me to use another type as execute's Response, which is the same as `GetDataResponse`. `GetDataResponse` is hard-coded in `GrpcWriter`. A bit change is needed